packages/wonder-blocks-layout/components/media-layout-context.test.js

```diff
@@ -0,0 +1,164 @@
+// @flow
+import React from "react";
+import {View} from "@khanacademy/wonder-blocks-core";
+
+import MediaLayout from "./media-layout.js";
+import MediaLayoutContext from "./media-layout-context.js";
+import {resizeWindow, matchMedia} from "../util/test-util.js";
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+
+import {
+    MEDIA_DEFAULT_SPEC,
+    MEDIA_INTERNAL_SPEC,
+    MEDIA_MODAL_SPEC,
+} from "../util/specs.js";
+
+describe("MediaLayoutContext", () => {
+    beforeEach(() => {
+        unmountAll();
+        window.matchMedia = matchMedia;
+    });
+
+    describe("overrideSize", () => {
+        it("should override the currentSize", async () => {
+            // Arrange
+            resizeWindow("large");
+            const promise = new Promise((resolve, reject) => {
+                mount(
+                    <MediaLayoutContext.Provider
+                        value={{
+                            overrideSize: "small",
+                            ssrSize: "large",
+                            mediaSpec: MEDIA_DEFAULT_SPEC,
+                        }}
+                    >
+                        <MediaLayout styleSheets={{}}>
+                            {({mediaSize, mediaSpec, styles}) => {
+                                resolve({mediaSize, mediaSpec, styles});
+                                return <View>Hello, world!</View>;
+                            }}
+                        </MediaLayout>
+                    </MediaLayoutContext.Provider>,
+                );
+            });
+
+            // Act
+            const args = await promise;
```

Comment:
What do you think about combining this with the `// Act` line since it really defines what the action is?

Reply:
I don't follow.  This is the `// Act` line. I can combine this with `// Arrange` and do something like:
```
// Arrange, Act
const args = await new Promise(...
```

Reply:
Yeah, that's what I was getting at. When the test reads like a story, it is often easier to understand what happens when.